plugins/org.eclipse.fordiac.ide.model.commands/src/org/eclipse/fordiac/ide/model/commands/change/ChangeStructCommand.java

```diff
@@ -19,22 +19,46 @@
 package org.eclipse.fordiac.ide.model.commands.change;
 
 import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.fordiac.ide.model.LibraryElementTags;
+import org.eclipse.fordiac.ide.model.data.DataType;
 import org.eclipse.fordiac.ide.model.data.StructuredType;
-import org.eclipse.fordiac.ide.model.helpers.PackageNameHelper;
 import org.eclipse.fordiac.ide.model.libraryElement.Demultiplexer;
 import org.eclipse.fordiac.ide.model.libraryElement.LibraryElementFactory;
 import org.eclipse.fordiac.ide.model.libraryElement.Multiplexer;
 import org.eclipse.fordiac.ide.model.libraryElement.StructManipulator;
-import org.eclipse.fordiac.ide.model.typelibrary.DataTypeLibrary;
+import org.eclipse.fordiac.ide.model.libraryElement.impl.ConfigurableFBManagement;
 
 public class ChangeStructCommand extends AbstractUpdateFBNElementCommand {
 
-	private final StructuredType newStruct;
+	private final DataType newStruct;
+	private final String newVisibleChildren;
+
+	public ChangeStructCommand(final StructManipulator mux) {
+		super(mux);
```

Comment:
yes but I didn't want to have the public constructor with "DataType", although I internally need to handle DataType because it could be an ErrorDataType which is then not a struct.... so I propose to wait for some tests and then possibly do further cleanups.